tests/ecmascript/test-bi-reflect-apply-construct.js

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Reflect.apply()
+ *  Reflect.construct()
+ */
+
+/*===
+Casper the friendly ghost
+maggie the eaty pig
+Darkling the undefined
+===*/
+
+function applyTest() {
+    var func = function(x, y) {
+        if (arguments.length >= 2) {
+            print(this.name, "the", x, y);
+        } else {
+            print(this.name, "the", x);
+        }
+    }
+    var casper = { name: "Casper" };
+    var maggie = { name: "maggie" };
+    var darkling = { name: "Darkling" };
+    Reflect.apply(func, casper, [ "friendly", "ghost" ]);
+    Reflect.apply(func, maggie, [ "eaty", "pig" ]);
+    Reflect.apply(func, darkling, []);
+}
+
+try {
+    applyTest();
```

Comment:
For coverage it'd be good to have an apply test for near-maximum number of supported arguments (say 250). This ensures that the implementation's `duk_require_stack()` calls are right (they are, but to ensure they keep on being right).